File: python/baggage.py
# This represents a single baggage record

# create table baggage(
#     id uuid,
#     ticket_id uuid,
#     passenger_id uuid,
#     passenger_name text
#     image text,
#     flight_id text, 
#     origin_city text, 
#     destination_city text,
#     carousel text,
#     PRIMARY KEY((ticket_id), passenger_id)
# );

class Baggage:
    def __init__(self, id, ticket_id, passenger_id, passenger_name, image, flight_id, origin_city, 
    destination_city, carousel):
        self.id = id
        self.ticket_id = ticket_id
        self.passenger_id = passenger_id
        self.passenger_name = passenger_name
        self.image = image
        self.flight_id = flight_id
        self.origin_city = origin_city
        self.destination_city = destination_city
        self.carousel = carousel


    def print(self):
        print("Baggage id: " + self.id + " for ticket: " + self.ticket_id + " for passenger " + self.passenger_name + " - From: " + self.origin_city + "  To: " + self.destination_city + 
            " is on carousel " + self.carousel)

    # Return this record as a single line in a CSV file
    def as_csv(self):
        return f"{self.id},{self.ticket_id},{self.passenger_id},{self.passenger_name},{self.image},{self.flight_id},{self.origin_city}," + \
            f"{self.destination_city},{self.carousel}\n"
        
    # Return this record as a single line in a CSV file
    @staticmethod
    def csv_header():
        return "id,ticket_id,passenger_id,passenger_name,image,flight_id,origin_city," + \
            "destination_city,carousel\n"

File: flutter_app/lib/entities/Baggage.dart
class Baggage {
  late String id;
  late String ticket_id;
  late String destination_city;
  late String origin_city;
  late String flight_id;
  late String image = "";
  late String passenger_id;
  late String passenger_name;
  late int carousel;

  Baggage() {}

  Baggage.fromJson(Map<String, dynamic> json) {
    id = json['id'] as String;
    ticket_id = json['ticket_id'] as String;
    destination_city = json['destination_city'] as String;
    origin_city = json['origin_city'] as String;
    flight_id = json['flight_id'] as String;
    image = json['image'] as String;
    passenger_id = json['passenger_id'] as String;
    passenger_name = json['passenger_name'] as String;
    carousel = json['carousel'] as int;
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['id'] = id;
    data['ticket_id'] = ticket_id;
    data['destination_city'] = destination_city;
    data['origin_city'] = origin_city;
    data['flight_id'] = flight_id;
    data['image'] = image;
    data['passenger_id'] = passenger_id;
    data['passenger_name'] = passenger_name;
    data['carousel'] = carousel;
    return data;
  }
}
